terraform/dev.tfvars

```diff
@@ -1,18 +1,22 @@
-aws_region             = "eu-west-2"
-az_count               = 2
-ecs_fargate_version    = "1.4.0"
-webapp_count           = 1
-webapp_fargate_cpu     = 256
-webapp_fargate_memory  = 512
-service_count          = 1
-service_fargate_cpu    = 256
-service_fargate_memory = 512
-db_storage             = 20
-db_max_storage         = 20
-db_delete_protection   = false
-db_instance_class      = "db.t2.micro"
-db_storage_encrypted   = false
-db_skip_final_snapshot = true
-webapp_client_id       = "1ccbe14d-00e3-43ac-a434-8f7a38e03366"
-service_api_id         = "5cdcbb41-958a-43b6-baa1-bbafd80b4f70"
-mca_azure_ad_tenant_id = "513fb495-9a90-425b-a49a-bc6ebe2a429e"
\ No newline at end of file
+aws_region                = "eu-west-2"
+az_count                  = 2
+ecs_fargate_version       = "1.4.0"
+webapp_count              = 1
+webapp_fargate_cpu        = 256
+webapp_fargate_memory     = 512
+service_count             = 1
+service_fargate_cpu       = 256
+service_fargate_memory    = 512
+db_storage                = 20
+db_max_storage            = 20
+db_delete_protection      = false
+db_instance_class         = "db.t2.micro"
+db_storage_encrypted      = false
+db_skip_final_snapshot    = true
+webapp_client_id          = "1ccbe14d-00e3-43ac-a434-8f7a38e03366"
+service_api_id            = "5cdcbb41-958a-43b6-baa1-bbafd80b4f70"
+mca_azure_ad_tenant_id    = "513fb495-9a90-425b-a49a-bc6ebe2a429e"
+next_auth_url             = "https://dev.406beacons.com"
+mca_azure_b2c_client_id   = "43557393-7cd2-416e-8bb2-96283dbdbcbc"
+mca_azure_b2c_tenant_name = "B2CMCGA"
+mca_azure_b2c_login_flow  = "B2C_1_login_beacons"
```

Comment:
Looking at these and even though we've been doing auth this sprint I'm having to juggle a lot in my head to work out the significance of these vars.

I'm wondering if there's a consistent naming convention we can set up.  Very difficult in this TF config given we're managing two services, with two different OAuth grant types, and two different providers.  

Some back-of-an-envelope options:

1.  `<service>-<grant type>-<variable name>`:

webapp-authorization_code-client_id        = "43557393-7cd2-416e-8bb2-96283dbdbcbc"
webapp-authorization_code-tenant_name = "B2CMCGA"
webapp-authorization_code-login_flow  = "B2C_1_login_beacons"
webapp-confidential_client-client_id          = "1ccbe14d-00e3-43ac-a434-8f7a38e03366"
service-confidential_client-api_id               = "5cdcbb41-958a-43b6-baa1-bbafd80b4f70"
webapp-next_auth_url             = "https://dev.406beacons.com"

2.  `<service>-<provider>-<variable name>`:

webapp-azure_b2c-client_id        = "43557393-7cd2-416e-8bb2-96283dbdbcbc"
webapp-azure_b2c-tenant_name = "B2CMCGA"
webapp-azure_b2c-login_flow  = "B2C_1_login_beacons"
webapp-azure_ad-client_id          = "1ccbe14d-00e3-43ac-a434-8f7a38e03366"
service-azure_ad-api_id               = "5cdcbb41-958a-43b6-baa1-bbafd80b4f70"
webapp-next_auth_url             = "https://dev.406beacons.com"

3.  Or, bigger change but I think in our long-term interest, separate out the webapp and service's own infrastructure to their respective repos while keeping only shared infra in this one.  This would solve the namespace issue.  

(I'd consider shared infra to be DNS, maybe some of the secrets manager resources, maybe the security groups that allow webapp and service to talk to each other.  Webapp and Service would not share an ECS cluster.)

Food for thought?

Reply:
Great spot, agreed!

I have arranged everything in the variables file that was logically grouped and having done that update it makes he head hurt a lot less and it's easier for me now to find variables. I have also done this in the `tfvars` files as well which I think helps with readability

What I have gone for your suggestion of:  `<service>_<provider - if applicable>_<variable name>`. Where service is either:

* Web facing form
* API

Provider is optional and would be defined to add additional context to the variable name, such as:

* Azure B2C
* Azure AD
* Next Auth

And then variable name is the what is this variable for, given the extra context, so:

* Client secret
* Tenant id
* JWT Secret

I went for option 2 as option 1 still feels like the names could be slightly ambiguous. You ideally want to ask the question: `What is the purpose of this variable` to be obvious from the name itself, with the description providing more detail if required.